daemon/volumes.go

```diff
@@ -366,3 +381,12 @@ func (container *Container) unmountVolumes() {
 		}
 	}
 }
+
+// unmountVolumesOnError is a convenient deferrable method which will unmount
```

Comment:
The container and volume `mount` and `unmount` methods were already separated, so I just put these `unmountOnErr` methods immediately after each of those.